### PR TITLE
Gp2 207 welcome back

### PR DESCRIPTION
**Pull request by bobmeredith**
To do (delete all that do not apply):

 - [x] Change has a jira ticket that has the correct status.
 - [x] Changelog entry added.
 - [x] Upgraded any vulnerable dependencies.
 - [x] Requirements have been compiled.

### CODE REVIEW

File: core/templates/core/dashboard.html [hunk 1]
```diff
@@ -1,9 +1,10 @@
 {% extends 'core/base.html' %}
+{% load to_json %}
```
Comment (webbyfox) [inline core/templates/core/dashboard.html:2]: Is this been used?

File: sso/models.py [hunk 2]
```diff
@@ -39,3 +39,19 @@ def save(self, *args, **kwargs):
     @cached_property
     def export_plan(self):
         return get_exportplan(self.session_id)
+
+    @cached_property
+    def user_profile(self):
+        return helpers.get_user_profile(self.session_id)
```
Comment (webbyfox) [inline sso/models.py:45]: wondering why they have used `session_id` as it implies dynamic for each session. Actual it is `sso_use_id`

File: tests/unit/sso/test_models.py [hunk 3]
```diff
@@ -1,13 +1,44 @@
...
+    assert user.update_user_profile({'firstname': 'Jim'}) == ok_result
+
+
+@mock.patch.object(sso_api_client.user, 'get_user_page_views')
+def test_has_visited_page(mock_get_user_page_views):
+    user = get_user()
+    page_views_response = {'result': 'ok', 'page_views': {'mypage': 1}}
+    mock_get_user_page_views.return_value = create_response(page_views_response)
+    assert user.get_page_views() == page_views_response
+    assert user.has_visited_page(page='mypage') == {'mypage': 1}
```
Comment (webbyfox) [inline tests/unit/sso/test_models.py:44]: Do we need a negative test? For example,  user not visited page and check `has_visited_page`

File: core/templates/core/dashboard.html (hunk 1 above)
Comment (bobmeredith) [reply core/templates/core/dashboard.html:2]: Fair point.
No it's not ATM, but it's a useful thing to show context while building up the page.
I'll take it out of this PR.

File: sso/models.py (hunk 2 above)
Comment (bobmeredith) [reply sso/models.py:45]: Yes, It is the sso_id, but the field in  BusinessSSOUser is called 'session_id'. It's static and unique to a user as it's the key used to retrieve the user profile from directory_sso.
I don't fancy doing a renaming exercise in this PR.

File: tests/unit/sso/test_models.py (hunk 3 above)
Comment (bobmeredith) [reply tests/unit/sso/test_models.py:44]: Could do.  The tests in directory_sso do negative tests, but there's no harm in one here too. 
Will add one - as a helper test though.
Essentially, the code in great-cms can either get all page views for this user/service and check for pages in the result, or filter the return to look at just one page. In the future, we may want to put the whole list in redis to speed things up.

Comment (webbyfox) [reply tests/unit/sso/test_models.py:44]: Yeah, I dont want as well but just observation.

File: sso/models.py (hunk 2 above)
Comment (webbyfox) [reply sso/models.py:45]: Yeah, I dont want as well but just observation.

**Comment by webbyfox** [comment]
LGTM, Just update the branch and good to merge

**Pull request merged**